instrumentation/http/src/main/java/brave/http/HttpServerRequest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.Propagation.Getter;
+
+/**
+ * Marks an interface for use in {@link HttpServerHandler#handleReceive(HttpServerRequest)}. This
+ * gives a standard type to consider when parsing an incoming context.
+ *
+ * @since 5.7
+ */
+public abstract class HttpServerRequest {
+  static final Getter<HttpServerRequest, String> GETTER = new Getter<HttpServerRequest, String>() {
+    @Override public String get(HttpServerRequest carrier, String key) {
+      return carrier.header(key);
+    }
+
+    @Override public String toString() {
+      return "HttpServerRequest::header";
+    }
+  };
+
+  /** Returns the underlying http request object. */
+  public abstract Object unwrap();
```

Comment:
Can consider `<T> T` so users don't have to cast explicitly

Reply:
if we do that, we can't covariant override, which I think we do somewhere. Plus `<T> T` is a little dodgy as it is less explicit that the type could be not what you think. instanceof is definitely safer right?

Reply:
rewrote the javadoc.